src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> byEmail = userService.findByEmail(email);
+        User user = byEmail.orElseThrow(()
+                -> new AuthenticationException("Incorrect password"));
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (hashedPassword.equals(user.getPassword())) {
+            return user;
+        } else {
+            throw new AuthenticationException("Incorrect password");
+        }
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (email == null || password == null || email.isEmpty() || password.isEmpty()) {
```

Comment:
I suggest to write method something like 'isBlank' and use it for password and email check